### ISSUE TITLE
ONNX Inference Problem

### ISSUE THREAD

**Issue by LopSdir** [RVC-Project/Retrieval-based-Voice-Conversion-WebUI#1983]
![image](https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/assets/49876859/5a53ab97-f648-4b38-b0be-f2e4d3566039)

I use the demo of inferencing by onnx, but I had this problem ,I try many times to solve this problem, but still can't solve it.
I also use the old onnx, but still can't.

So may I know how to solve this problem or can I get the testing onnx file

**Comment by fumiama** [comment]
Try the latest commit.

**Comment by yc930401** [comment]
![image](https://github.com/user-attachments/assets/fae0612b-dbb2-43ad-b2a2-4f8373c6f5d8)

最新版本代码，还是同样的问题。

**Comment by samolego** [comment]
I too get this error (with different expected and requested sizes) :cry:

**Comment by zhixingheyixsh** [comment]
我也遇到这个问题了，看报错是在 /enc_p/encoder/attn_layers.0/Reshape_10，应该是infer/lib/infer_pack/attentions.py 中Encoder 中的 Mutiheadattention 里有个reshape,但是我看forward里是没有的，反而上一层 TextEncoder 的forward里有几个reshape.

另外，请问你的vec-768-layer-12.onnx 是在哪儿块下载的？是这个地址吗 https://huggingface.co/NaruseMioShirakana/MoeSS-SUBModel

其它地方也有这个问题，可以参考#2298 
有个地方说了解决方案
#1830

**Comment by chuckcxr** [comment]
> I also encountered this problem. The error is in /enc_p/encoder/attn_layers.0/Reshape_10. There should be a reshape in Mutiheadattention in Encoder in infer/lib/infer_pack/attentions.py, but I don't see it in forward. Instead, there are several reshapes in the forward of the previous layer TextEncoder.
> 
> Also, where did you download vec-768-layer-12.onnx? Is it this address https://huggingface.co/NaruseMioShirakana/MoeSS-SUBModel
> 
> This problem also occurs in other places. You can refer to [#2298.](https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/issues/2298) There is a solution mentioned in [#1830.](https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/issues/1830)

False....that link they point to this page.....this page point to that page.....circles and circles and no answer >_<

**Comment by chen-Mengjiang** [comment]
一般是python版本不兼容问题，导致你的张量数据不对，你用VSCode设置python的时候直接选包里runtime的试试。我已经实现参照onnx_inference_demo.py实现了C++的推理，但是感觉效果不满意，还在调。如果你也是C++的，可以交流一下
